.circleci/config.yml

```diff
@@ -76,65 +92,56 @@ jobs:
                 exit 1
               fi
 
-      - checkout
-      - *post_checkout
-      - run:
-          name: Skip screenshotter if only documentation has been changed
-          command: |
-              if [[ $CHANGED ]]; then
-                echo "$CHANGED" | grep -qvE '^docs/|^LICENSE|\.md$' || circleci step halt
-              fi
-
-      - *restore_node_modules_cache
-      - *yarn_install
-      - save_cache:
-          key: *node_modules_cache_key
-          paths:
-            - node_modules
+      - checkout_repo
+      - skip_if_only_changed:
+          filter: '^docs/|^LICENSE|\.md$'
+      - install_dependencies
 
       - add_ssh_keys:
           fingerprints:
             - "e1:ac:13:98:98:8b:fd:38:81:12:28:55:75:a1:da:73"
       - run:
           name: Greenkeeper - update and commit yarn.lock
+          environment:
+            # greenkeeper-lockfile detects build by checking it's the first job
+            # of the first workflow. As we run jobs parallel, it may not be the
+            # first job
+            CIRCLE_PREVIOUS_BUILD_NUM: ""
           command: |
               ./node_modules/.bin/greenkeeper-lockfile-update
               ./node_modules/.bin/greenkeeper-lockfile-upload
 
       - run:
-          name: Run tests
-          command: yarn test --coverage
-      - *codecov
-
+          name: Lint code
+          command: yarn test:lint
+      - skip_if_only_changed:
+          filter: '^static/|^website/'
       - run:
-          name: Build KaTeX
-          command: USE_TTF=false yarn build
-
-      - store_artifacts:
-          path: dist/katex.min.js
-          destination: katex.min.js
-      - store_artifacts:
-          path: dist/katex.min.css
-          destination: katex.min.css
-      - store_artifacts:
-          path: dist/fonts
-          destination: fonts
+          name: Run Flow and Jest tests
+          command: |
+              yarn test:flow
+              yarn test:jest --coverage
+      - codecov:
+          flag: test
 
   firefox:
     docker:
       - image: circleci/node:6
       - image: selenium/standalone-firefox:2.48.2
-    <<: *screenshotter
+    steps:
```

Comment:
How does "stacking" of images work?  I'm guessing that `selenium/standalone-firefox:2.48.2` doesn't include `node` and yet the `screenshotter` command runs `node`.

Reply:
@kevinbarabash

> It is possible to specify multiple images for your job. Specify multiple images if, for example, you need to use a database for your tests or for some other required service. In a multi-image configuration job, all steps are executed in the container created by the first image listed. All containers run in a common network and every exposed port will be available on localhost from a primary container.

Reply:
I was thinking about how I might use this for `vite`, but since there's a node script that needs to run from inside the container it uses even when running it locally I needed to include `node` in the custom container.  I was thinking of using one of circleci's so-called "browsers" images that include Chrome and Firefox and starts Xvfb... then I'd only need to download the `vite` code.